tests/unit/package_managers/test_pip.py

```diff
@@ -54,10 +57,14 @@ def rooted_tmp_path(tmp_path: Path) -> RootedPath:
 @pytest.mark.parametrize("cfg_exists", [True, False])
 @pytest.mark.parametrize("cfg_name", ["name_in_setup_cfg", None])
 @pytest.mark.parametrize("cfg_version", ["version_in_setup_cfg", None])
+@pytest.mark.parametrize("repo_exists", [True, False])
+@pytest.mark.parametrize("origin", ["bar-subpath", None])
```

Comment:
Do these combinations make sense?
* repo_exists=True, origin=None
* repo_exists=False, origin="bar-subpath"

It looks like `origin` by itself covers all the cases we need